burgerSync.js

```diff
@@ -0,0 +1,252 @@
+const WebSocket = require("ws");
+const requester = require('request-promise');
+
+class BurgerSync {
+    constructor(port, burgerNode) {
+        this.burgerNode = burgerNode;
+
+        this.peers = {};
+        this.MESSAGE_TYPE = {
+            INITIAL_HANDSHAKE_QUERY: 0,
+            INITIAL_HANDSHAKE_RESPONSE: 1,
+            REQUEST_SYNC_CHAIN: 2,
+            RESPONSE_SYNC_CHAIN: 3,
+            BROADCAST_NEW_BLOCK: 4,
+            BROADCAST_NEW_TRANSACTION: 5,
+            INVALID_REQUEST: 6
+        }
+
+        this.webSocket = new WebSocket.Server({ port });
+        this.webSocket.on('connection', this.initializeConnection.bind(this));
```

Comment:
@24thsaint why are you binding to `this` when it is `this` that is already making the call?

Reply:
Hahaha, I have no idea too, but without the bind, calls to `this` become undefined. Javascript quirks.

Reply:
I guess `call` would've made more sense to me than `bind` here in that case, but I do agree with you that it's strange that you can't just pass in the function to be called.